consensus/fork_choice/src/fork_choice.rs

```diff
@@ -412,21 +412,25 @@ where
             AttestationShufflingId::new(anchor_block_root, anchor_state, RelativeEpoch::Next)
                 .map_err(Error::BeaconStateError)?;
 
-        // Default any non-merge execution block hashes to 0x000..000.
-        let execution_status = anchor_block.message_merge().map_or_else(
-            |()| ExecutionStatus::irrelevant(),
-            |message| {
-                let execution_payload = &message.body.execution_payload;
-                if execution_payload == &<_>::default() {
-                    // A default payload does not have execution enabled.
-                    ExecutionStatus::irrelevant()
-                } else {
-                    // Assume that this payload is valid, since the anchor should be a trusted block and
-                    // state.
-                    ExecutionStatus::Valid(message.body.execution_payload.block_hash())
-                }
-            },
-        );
+        let execution_status = anchor_block
+            .message()
+            .body()
+            .execution_payload()
+            .map_or_else(
+                // If the block doens't have an execution payload then it can't
+                // have execution enabled.
+                |_| ExecutionStatus::irrelevant(),
+                |execution_payload| {
+                    if execution_payload == &<_>::default() {
```

Comment:
```suggestion
                    if execution_payload.is_default_with_empty_roots() {
```

Reply:
This method isn't available here since we're using a `SignedBeaconBlock` with `FullPayload`.

Reply:
Oh wait, it might be because this is based on `unstable`. One sec.

Reply:
We get a FullPayloadRef here which doesn't have a `Default` impl

Reply:
Yep, my bad. I read the macros wrong. I went with `is_default_with_empty_roots` since that seems to be what we're standardizing on. It shouldn't make a difference since this is fixed to `FullPayload`, but it seems like good practice.

Reply:
> We get a FullPayloadRef here which doesn't have a `Default` impl

Yep, this became true when I rebased it onto `capella` 👍